diff --git a/cmake/OpenCVFindXimea.cmake b/cmake/OpenCVFindXimea.cmake
index 2d93292..6daf9a7 100644
--- a/cmake/OpenCVFindXimea.cmake
+++ b/cmake/OpenCVFindXimea.cmake
@@ -37,8 +37,19 @@ elseif(APPLE)
   else()
     set(XIMEA_FOUND 0)
   endif()
+elseif(APPLE)
+  if(EXISTS /Library/Frameworks/m3api.framework)
+    set(XIMEA_FOUND 1)
+  else()
+    set(XIMEA_FOUND 0)
+  endif()
 else()
-  if(EXISTS /opt/XIMEA)
+  if(EXISTS ${CMAKE_INSTALL_PREFIX}/opt/XIMEA)
+    set(XIMEA_FOUND 1)
+    # set folders
+    set(XIMEA_PATH ${CMAKE_INSTALL_PREFIX}/opt/XIMEA/include)
+    set(XIMEA_LIBRARY_DIR ${CMAKE_INSTALL_PREFIX}/lib)
+  elseif(EXISTS /opt/XIMEA)
     set(XIMEA_FOUND 1)
     # set folders
     set(XIMEA_PATH /opt/XIMEA/include)
diff --git a/modules/highgui/include/opencv2/highgui/highgui_c.h b/modules/highgui/include/opencv2/highgui/highgui_c.h
index 1f86abb..c73375e 100644
--- a/modules/highgui/include/opencv2/highgui/highgui_c.h
+++ b/modules/highgui/include/opencv2/highgui/highgui_c.h
@@ -444,6 +444,7 @@ enum
     CV_CAP_PROP_XI_AG_MAX_LIMIT  = 418,      // Maximum limit of gain in AEAG procedure
     CV_CAP_PROP_XI_AEAG_LEVEL    = 419,       // Average intensity of output signal AEAG should achieve(in %)
     CV_CAP_PROP_XI_TIMEOUT       = 420,       // Image capture timeout in milliseconds
+    CV_CAP_PROP_XI_PRM_ACQ_TIMING_MODE = 421, // timing mode
 
     // Properties for Android cameras
     CV_CAP_PROP_ANDROID_FLASH_MODE = 8001,
diff --git a/modules/highgui/src/cap_ximea.cpp b/modules/highgui/src/cap_ximea.cpp
index 13a4538..0a152d0 100644
--- a/modules/highgui/src/cap_ximea.cpp
+++ b/modules/highgui/src/cap_ximea.cpp
@@ -209,10 +209,10 @@ void CvCaptureCAM_XIMEA::resetCvImage()
         if(frame) cvReleaseImage(&frame);
         frame = NULL;
 
-        switch( image.frm)
+        switch(format)
         {
         case XI_MONO8       :
-        case XI_RAW8        : frame = cvCreateImage(cvSize( image.width, image.height), IPL_DEPTH_8U, 1); break;
+        case XI_RAW8        : frame = cvCreateImage(cvSize( image.width, image.height), IPL_DEPTH_8U, 1); printf(" NEWWWW\n"); break;
         case XI_MONO16      :
         case XI_RAW16       : frame = cvCreateImage(cvSize( image.width, image.height), IPL_DEPTH_16U, 1); break;
         case XI_RGB24       :
@@ -263,6 +263,7 @@ double CvCaptureCAM_XIMEA::getProperty( int property_id )
     case CV_CAP_PROP_XI_AE_MAX_LIMIT  : xiGetParamInt( hmv, XI_PRM_EXP_PRIORITY, &ival); return ival;
     case CV_CAP_PROP_XI_AG_MAX_LIMIT  : xiGetParamFloat( hmv, XI_PRM_AG_MAX_LIMIT, &fval); return fval;
     case CV_CAP_PROP_XI_AEAG_LEVEL    : xiGetParamInt( hmv, XI_PRM_AEAG_LEVEL, &ival); return ival;
+    case CV_CAP_PROP_XI_PRM_ACQ_TIMING_MODE : xiGetParamInt( hmv, XI_PRM_ACQ_TIMING_MODE, &ival); return ival;
     case CV_CAP_PROP_XI_TIMEOUT       : return timeout;
 
     }
@@ -307,9 +308,13 @@ bool CvCaptureCAM_XIMEA::setProperty( int property_id, double value )
     case CV_CAP_PROP_XI_AE_MAX_LIMIT  : mvret = xiSetParamInt( hmv, XI_PRM_EXP_PRIORITY, ival); break;
     case CV_CAP_PROP_XI_AG_MAX_LIMIT  : mvret = xiSetParamFloat( hmv, XI_PRM_AG_MAX_LIMIT, fval); break;
     case CV_CAP_PROP_XI_AEAG_LEVEL    : mvret = xiSetParamInt( hmv, XI_PRM_AEAG_LEVEL, ival); break;
+    case CV_CAP_PROP_XI_PRM_ACQ_TIMING_MODE :  mvret = xiSetParamInt( hmv, XI_PRM_ACQ_TIMING_MODE, ival); break;
     case CV_CAP_PROP_XI_TIMEOUT       : timeout = ival; break;
     }
 
+    if (property_id == CV_CAP_PROP_XI_DATA_FORMAT) 
+        resetCvImage();
+
     if(mvret != XI_OK)
     {
         errMsg("Set parameter error", mvret);
